packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1991,6 +1999,55 @@ export class ContainerRuntime
 		this.removeAllListeners();
 	}
 
+	/**
+	 * Utility function to check if any blobs under a snapshot tree is missing and if so, then set
+	 * omitted flag to true for that snapshot tree.
+	 * @param snapshotTree - snapshotTree to be evaluated for missing blobs.
+	 * @param blobContents - blobContents of the snapshot.
+	 */
+	private evaluateSnapshotForOmittedBlobContents(
+		snapshotTree: ISnapshotTree,
+		blobContents: Map<string, ArrayBuffer>,
+	) {
+		for (const [_, id] of Object.entries(snapshotTree.blobs)) {
+			if (!blobContents.has(id)) {
+				snapshotTree.omitted = true;
+				break;
+			}
+		}
+		for (const [_, treeChild] of Object.entries(snapshotTree.trees)) {
+			this.evaluateSnapshotForOmittedBlobContents(treeChild, blobContents);
+		}
+	}
+
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IContainerRuntimeBase.isSnapshotFetchRequired}
+	 */
+	public isSnapshotFetchRequired(pathParts: string[]): boolean {
+		validatePrecondition(
+			this.snapshotWithContents !== undefined,
+			"Use this api in case we fetched the complete snapshot along with the blob contents",
+		);
+		// Find the snapshotTree inside the snapshot based on the path as given in the request.
+		const hasIsolatedChannels = rootHasIsolatedChannels(this.metadata);
+		const snapshotTreeForPath = this.getSnapshotTreeForPath(
+			this.snapshotWithContents.snapshotTree,
+			pathParts,
+			hasIsolatedChannels,
+		);
+		if (snapshotTreeForPath === undefined) {
+			throw new UsageError("snapshotTree not found for given path", {
+				details: JSON.stringify({ pathPartsLength: pathParts.length }),
+			});
+		}
+		for (const [_, id] of Object.entries(snapshotTreeForPath.blobs)) {
+			if (!this.snapshotWithContents.blobContents.has(id)) {
```

Comment:
Do we have a way of checking all our blobContents? What if at some point the fetch isn't required because we have the blobId because of a groupId fetch we already made? Or am I missing something in my thinking here?

Reply:
I think you are correct, we can update our blobs list on a groupid fetch.

Reply:
.

Reply:
We do cache the grouped snapshot in container runtime for some time with an expiry, so that if fetched again before expiry, we don't end up making network call again.